AEM/tf-itex-cheat-sheet.md

```diff
@@ -1,244 +1,73 @@
-
-Toggle a GPU Back End (in the Code): Set by Default
-
-itex.set_backend(‘GPU’)
-
-Toggle a GPU Back End (Command Prompt): Set by Default
-
-Export ITEX_XPU_BACKEND="GPU"
-
-Advanced Automatic Mixed Precision (in the Code): A Basic Configuration with Improved Inference Speed with Reduced Memory Consumption
-
-auto_mixed_precision_options = itex.AutoMixedPrecisionOptions()
-
-
-
-auto_mixed_precision_options.data_type = itex.BFLOAT16 # or itex.FLOAT16
-
-Advanced Automatic Mixed Precision (Command Prompt): A Basic Configuration with Improved Inference Speed with Reduced Memory Consumption
-
-export ITEX_AUTO_MIXED_PRECISION=1 
-
-export ITEX_AUTO_MIXED_PRECISION_DATA_TYPE="BFLOAT16" # or "FLOAT16"
-
-Customized AdamW Optimizer (in the Code)
-
-itex.ops.AdamWithWeightDecayOptimizer(
-
-    weight_decay_rate=0.001,
-
-    learning_rate=0.001, beta_1=0.9,
-
-    beta_2=0.999,
-
-    epsilon=1e-07, name='Adam',
-
-    exclude_from_weight_decay=["LayerNorm",
-
-    "layer_norm", "bias"], **kwargs
-
-)
-
-
-
-Customized Layer Normalization (in the Code)
-
-itex.ops.LayerNormalization(
-
-    axis=-1, epsilon=0.001, center=True,
-
-    scale=True,
-
-    beta_initializer='zeros',
-
-    gamma_initializer='ones',
-
-    beta_regularizer=None,
-
-    gamma_regularizer=None,
-
-    beta_constraint=None,
-
-    gamma_constraint=None, **kwargs
-
-)
-
-Customized GELU (in the Code)
-
-itex.ops.gelu(
-
-    features, approximate=False, name=None
-
-)
-
-Customized LSTM (in the Code)
-
-itex.ops.ItexLSTM(
-
-    200, activation='tanh',
-
-    recurrent_activation='sigmoid',
-
-    use_bias=True,
-
-    kernel_initializer='glorot_uniform',
-
-    recurrent_initializer='orthogonal',
-
-    bias_initializer='zeros', **kwargs
-
-)
-
-
-
+<br><br>
+## Intel® Optimization for TensorFlow*: A Public Release from Google
+Features and optimizations for TensorFlow* on Intel hardware are frequently upstreamed and included in stock TensorFlow* releases. As of TensorFlow* v2.9, Intel® oneAPI Deep Neural Network Library (oneDNN) optimization is automatically enabled.
+
+For more information, see [TensorFlow](https://www.tensorflow.org/).
+
+<table><tbody>
+<tr><td>Basic Installation Using PyPI*</td><td>pip install tensorflow</td></tr>
+<tr><td>Basic Installation Using Anaconda*</td><td>conda install -c conda-forge tensorflow</td></tr>
+<tr><td>Import TensorFlow</td><td>import tensorflow as tf</td></tr>
+<tr><td>Capture a Verbose Log (Command Prompt)</td><td>export ONEDNN_VERBOSE=1</td></tr>
+<tr><td>Parallelize Execution (in the Code)</td><td>tf.config.threading.set_intra_op_parallelism_threads(&lt;number of physical cores per socket&gt;) <br> tf.config.threading.set_inter_op_parallelism_threads(&lt;number of sockets&gt;) <br> tf.config.set_soft_device_placement(True)</td></tr>
+<tr><td>Parallelize Execution (Command Prompt)</td><td>export TF_NUM_INTRAOP_THREADS=&lt;number of physical cores per socket&gt; <br> export TF_NUM_INTEROP_THREADS=&lt;number of sockets&gt;</td></tr>
```

Comment:
does it work for all TF workloads? it might be supported only for some TF workloads like TF benchmark.

Reply:
You are right that sometimes these recommended settings might not give the best perf in some workloads. Sometimes settting INTRA to logical cores per socket might bring better perf. Adding a note to tune the para according to workloads. Is it okay?

Reply:
hi, the question is for TF_NUM_INTEROP_THREADS. Does TF itself get those Env variables or does workload get env variable instead?